src/lib/dom-api-shady.html

```diff
@@ -422,6 +422,47 @@
 
   Object.defineProperties(DomApi.prototype, {
 
+    activeElement: {
+      get: function() {
+        var active = document.activeElement;
+        if (!active) {
+          return null;
+        }
+        var isShadyRoot = !!this.node._isShadyRoot;
+        if (this.node !== document) {
+          // If this node isn't a document or shady root, then it doesn't have
+          // an active element.
+          if (!isShadyRoot) {
+            return null;
+          }
+          // If this shady root's host is the active element or the active
+          // element is not a descendant of the host (in the composed tree),
+          // then it doesn't have an active element.
+          if (this.node.host === active ||
+              !this.node.host.contains(active)) {
+            return null;
+          }
+        }
+        // This node is either the document or a shady root of which the active
+        // element is a (composed) descendant of its host; iterate upwards to
+        // find the active element's most shallow host within it.
+        var activeRoot = Polymer.dom(active).getOwnerRoot();
+        while (activeRoot && activeRoot !== this.node) {
+          active = activeRoot.host;
+          activeRoot = Polymer.dom(active).getOwnerRoot();
```

Comment:
The convention in this module is to use `dom` which is locally mapped to `Polymer.dom`.